packages/contracts/contracts/core/SphinxModule.sol

```diff
@@ -34,7 +34,7 @@ contract SphinxModule is ReentrancyGuard, Enum, ISphinxModule, Initializable {
     /**
      * @inheritdoc ISphinxModule
      */
-    string public constant override VERSION = "1.0.0";
+    string public constant override VERSION = "1.0.0"; // TODO(later): what version?
```

Comment:
@RPate97 

Here's the new Sphinx Module. This design prioritizes simplicity and backwards compatibility. I haven't cleaned up this contract yet since I wanted to get your feedback on the design decisions first. I also haven't attempted to break the new contract yet.

### Cancellations

The key difference with this Module is that cancellations now occur in the `approve` function instead of a `cancel` function. I chose this design because it simplifies the Sphinx Module and Merkle tree logic.

It's necessary for the new Module to handle the following edge case:
1. Say Deployment 1 contains Transactions A, B, and C.
2. Say Transaction A is executed, but Transaction B reverts, causing the deployment to stall.
3. User signs Deployment 2, which cancels Deployment 1 and executes Transactions B and C.
4. Transaction B from Deployment 1 is executed successfully.
5. Deployment 2 is approved on-chain and executed successfully. This means Transaction B is executed twice, which is potentially dangerous.

This edge case is possible because deployments aren't explicitly marked as `FAILED` on-chain anymore. Notably, the Merkle root nonce is insufficient to prevent this scenario.

In the new design, we resolve this scenario by replacing the `merkleRootNonce` with two new fields in the `APPROVE` leaf: `previousMerkleRoot` and `previousLeavesExecuted`. A new Merkle root can only be approved if:
- The `previousMerkleRoot` matches the `latestMerkleRoot` (which is a new state variable in the Module).
- The `previousLeavesExecuted` matches the number of leaves executed in the `previousMerkleRoot`.

It'll be easy to collect these two fields during the proposal's collection phase.

These fields are essentially a more granular `merkleRootNonce`, so I removed that field because it's not necessary anymore.

### Pre-flight checks

You'll notice that the new Module doesn't contain any logic for pre-flight checks. I'd prefer if pre-flight checks are implemented as `EXECUTE` actions that are executed within the Gnosis Safe. This decreases the complexity of the new Module and allows us to change the implementation of the pre-flight checks without re-auditing the Module. This design decision means that deployments can be active without being executable yet. I'm not sure if this meaningfully increases the complexity of the executor.

### `activeMerkleRoot` -> `latestMerkleRoot`

I replaced the `activeMerkleRoot` state variable with the `latestMerkleRoot` state variable to support the cancellation design described above. The only difference between these two state variables is that the `latestMerkleRoot` is never set to `bytes32(0)`, whereas the `activeMerkleRoot` was set to `bytes32(0)` when a deployment was completed.